src/main/java/org/opensearch/knn/jni/PlatformUtils.java

```diff
@@ -80,4 +81,38 @@ public static boolean isAVX2SupportedBySystem() {
         }
         return false;
     }
+
+    public static boolean isAVX512SupportedBySystem() {
+
+        if (!Platform.isIntel() || Platform.isMac() || Platform.isWindows()) {
```

Comment:
does mac support avx2 but not avx512?

Reply:
that's right. mac moved away from x86 right around the time avx512 was introduced. 
I don't have mac systems to verify this, and @naveentatikonda confirmed his mac system doesn't have avx512. 
if you know of any mac systems supporting avx512, I can update the check here.